src/domain/entities/MetadataItem.ts

```diff
@@ -55,12 +55,18 @@ export interface DataDimensionItem {
     programIndicator?: Ref;
 }
 
-export type MetadataItem = Ref & SharedObject & { [key: string]: any | undefined };
+export type MetadataItem = Ref & {
+    sharing: {
+        userGroups: Record<string, SharingSetting>;
+        users: Record<string, SharingSetting>;
+        public: string;
+    };
+} & SharedObject & { [key: string]: any | undefined };
 
 export function isValidModel(model: string): model is MetadataModel {
     return ["dataSets", "programs", "dashboards"].includes(model);
 }
 
 export function isValidMetadataItem(item: any): item is MetadataItem {
-    return item.id && item.publicAccess && item.userAccesses && item.userGroupAccesses;
+    return item.id && item.sharing;
```

Comment:
To make it retro-compatible:  item should have or sharing object or the other properties